experimental/GModule/Cohomology.jl

```diff
@@ -474,7 +474,7 @@ export action, cohomology_group, extension, pc_group_with_isomorphism
 export induce, is_consistent, istwo_cocycle, all_extensions
 export split_extension, extension_with_abelian_kernel
 
-Oscar.dim(C::GModule) = rank(C.M)
+Oscar.dim(C::GModule) = torsion_free_rank(C.M)
```

Comment:
```suggestion
Oscar.dim(C::GModule) = rank(C.M)
```
This delegates to the module and not the group. So the change here is wrong

Reply:
I'm not sure this is true. `C.M` is "the module", but there are some signatures with `C::GModule{..., FinGenAbGroup}` and the second type parameter is for `C.M` as far as I can see. Somebody who knows about the code should have a look.

Reply:
Oh, I think it can be both here. So we would need something like
```julia
Oscar.dim(C::GModule{..., FinGenAbGroup}) = torsion_free_rank(C.M)
Oscar.dim(C:GModule) = rank(C.M)
```